pkg/cri/resource-manager/agent/agent.go

```diff
@@ -55,13 +61,20 @@ type Interface interface {
 
 // agentInterface implements Interface
 type agentInterface struct {
-	cli agent_v1.AgentClient
+	socket string
+	cli    agent_v1.AgentClient
 }
 
 // NewAgentInterface connects to cri-resource-manager-agent gRPC server
 // and return a new Interface
 func NewAgentInterface(socket string) (Interface, error) {
-	a := &agentInterface{}
+	a := &agentInterface{
+		socket: socket,
+	}
+
+	if a.IsDisabled() {
+		return a, nil
+	}
```

Comment:
Just thinking aloud: my first thought was that should we even call `NewAgentInterface` it it's disabled 🤔 That would of course mean doing `if agent != nil` checks elsewhere in the code base (instead of `IsDisabled()` checks in the agentinterface methods). Matybe it's safer this way.

Another point in this pondering is that I don't think the static pools policy works without agent(?) And it prolly shouldn't start at all 🧐

Reply:
I wanted to make this as little intrusive as possible, hence the IsDisabled() checks.

Reply:
Updated `static-pools` to bail out (also) if agent is disabled.